kore/src/Kore/Simplify/Condition.hs

```diff
@@ -168,15 +177,33 @@ simplifyPredicates sideCondition original = do
     let predicates =
             SideCondition.simplifyConjunctionByAssumption original
                 & fst . extract
-    simplifiedPredicates <-
+    simplifiedPredicates <- do
+        let eliminatedExists =
+                map (simplifyPredicateExistElim $ freeVariableNames original <> freeVariableNames sideCondition) $
+                    toList predicates
         simplifyPredicatesWithAssumptions
             sideCondition
-            (toList predicates)
+            eliminatedExists
     let simplified = foldMap mkCondition simplifiedPredicates
     if original == simplifiedPredicates
         then return (Condition.markSimplified simplified)
         else simplifyPredicates sideCondition simplifiedPredicates
 
+{- | Simplify an existential predicate by removing the existential binder
+under the assumption that the others are true.
```

Comment:
> under the assumption that the others are true

Not sure what this refers to?

Reply:
sorry, i think that was a copy-paste from the comment below